internal/middleware/logging.go

```diff
@@ -42,25 +74,33 @@ func NewStreamingLoggingInterceptor(logger logger.Logger) grpc.StreamServerInter
 	return func(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
 		start := time.Now()
 
-		err := handler(srv, stream)
+		fields := []zap.Field{zap.String(methodKey, info.FullMethod)}
 
-		fields := []zap.Field{
-			zap.String("method", info.FullMethod),
-			zap.Duration("took", time.Since(start)),
+		if requestID, ok := RequestIDFromContext(stream.Context()); ok {
+			fields = append(fields, zap.String(requestIDKey, requestID))
 		}
 
+		spanCtx := trace.SpanContextFromContext(stream.Context())
+		if spanCtx.HasTraceID() {
+			fields = append(fields, zap.String(traceIDKey, spanCtx.TraceID().String()))
+		}
+
+		err := handler(srv, stream)
+
+		fields = append(fields, zap.Duration(reqDurationKey, time.Since(start)))
```

Comment:
<img width="680" alt="Screen Shot 2023-02-03 at 2 20 18 PM" src="https://user-images.githubusercontent.com/2899204/216713066-3f53a530-b73b-4774-8c81-fa0997d3f708.png">

I see cases where request duration is `0ms`. We should investigate.

----

Also, bringing this up for further conversation for the maintainer team..

We should be more methodical about the observability componentry in OpenFGA. We have traces, metrics, and logs. The way we capture `req_duration_ms` will lead to differences in reported values as compared to the traces and metrics. For example, the traces and metrics wrap the endpoints early on in the middleware chain and this logger middleware wraps the endpoints after those middleware chains.

I'm concerned that we're reporting values across 3 different components (traces, metrics, and logs) and we should be more strategic about how we report latency based metrics in OpenFGA. I would prefer to reserve this metric for custom request duration metrics and traces. Log based metrics on this field aren't a best practice IMO.

Reply:
I think the `0ms` request duration is because it actually took less than `0.5ms` from the pov of that middleware. E.g. create store in memory gives something like:
<img width="1903" alt="Screen Shot 2023-02-03 at 14 28 39" src="https://user-images.githubusercontent.com/28915792/216723168-5656b32c-2042-4da1-a84e-2c2388ad9ad4.png">

Reply:
Perhaps the default of fractions of a second is better?

Reply:
Yeah, if we lose resolution then I would go that route. It doesn't make sense to have a `0ms` value there.

Reply:
Also just noting that this discrepancy is "exhibit a" of my point above 😄 

> The way we capture `req_duration_ms` will lead to differences in reported values as compared to the traces and metrics.

As we can see, we get `187us` precision in the trace and we see `0ms` in the log. That's misleading at best.

Reply:
Agreed 💯